src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs

```diff
@@ -3650,6 +3650,35 @@ partial interface IGoo<T, U> { }
             Assert.Equal("T1", param.Name);
         }
 
+        [Fact]
+        public void PartialMethodRenameParameters()
+        {
+            var text = @"namespace NS
+{
+    public partial class MyClass
+    {
+        partial void F<T, U>(T t) where T : class;
+        partial void F<T, U>(T tt) where T : class {}
+    }
+}";
+            var comp = CreateCompilation(text);
+            comp.VerifyDiagnostics(
+                // (6,22): warning CS8826: Partial method declarations 'void MyClass.F<T, U>(T t)' and 'void MyClass.F<T, U>(T tt)' have differences in parameter names, parameter types, or return types.
+                //         partial void F<T, U>(T tt) where T : class {}
+                Diagnostic(ErrorCode.WRN_PartialMethodTypeDifference, "F").WithArguments("void MyClass.F<T, U>(T t)", "void MyClass.F<T, U>(T tt)").WithLocation(6, 22)
+                );
+
+            var ns = comp.SourceModule.GlobalNamespace.GetMembers("NS").Single() as NamespaceSymbol;
+            var type1 = ns.GetTypeMembers("MyClass").Single() as NamedTypeSymbol;
+            Assert.Equal(0, type1.TypeParameters.Length);
+            var f = type1.GetMembers("F").Single() as MethodSymbol;
+            Assert.Equal(2, f.TypeParameters.Length);
```

Comment:
>TypeParameters [](http://example.com/codeflow?start=30&length=14)

Should these checks be using `Parameters` instead?

Reply:
@cston The number of parameters is 1, not 2. So this change will fail the test.

Reply:
> The number of parameters is 1, not 2. So this change will fail the test.

I meant that any asserts should probably compare `Parameters` rather than `TypeParameters` since, unlike the following test, the parameter names are distinct not the type parameter names.